oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/oidc/user/DefaultOidcUser.java

```diff
@@ -51,34 +51,68 @@ public class DefaultOidcUser extends DefaultOAuth2User implements OidcUser {
 	 * @param authorities the authorities granted to the user
 	 * @param idToken the {@link OidcIdToken ID Token} containing claims about the user
 	 */
+	@Deprecated
 	public DefaultOidcUser(Collection<? extends GrantedAuthority> authorities, OidcIdToken idToken) {
 		this(authorities, idToken, IdTokenClaimNames.SUB);
 	}
 
+	/**
+	 * Constructs a {@code DefaultOidcUser} using the provided parameters.
+	 * @param idToken the {@link OidcIdToken ID Token} containing claims about the user
+	 * @param authorities the authorities granted to the user
+	 */
+	public DefaultOidcUser(OidcIdToken idToken, Collection<? extends GrantedAuthority> authorities) {
+		this(null, idToken, authorities);
```

Comment:
Is this correct? It seems like it should be `idToken.getAttribute(IdTokenClaimNames.SUB)`.

Same question for `DefaultOidcUser(OidcIdToken, OidcUserInfo, Collection<? extends GrantedAuthority>)`

Reply:
Both constructors ultimately delegate to

```java
public DefaultOAuth2User(String name, Map<String, Object> attributes,
			Collection<? extends GrantedAuthority> authorities) {
    // ...
    this.name = (name != null) ? name : (String) this.attributes.get("sub");
}
```

which is consistent with the constructor behavior of `DefaultOAuth2AuthenticatedPrincipal`, although using `IdTokenClaimNames.SUB` instead of `"sub"` would be more appropriate.

Reply:
However, for that to succeed I'd still need to remove the `Assert.notNull(name, "name cannot be null");` from the constructor which is still present at the moment.